### PR TITLE
feat: idam stub support (CMC-894)

### PR DESCRIPTION
**Pull request by pd-dudkiewicz**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CMC-894

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: ccd-definition/AuthorisationCaseState/systemUpdate.json [hunk 1]
```diff
@@ -20,13 +20,6 @@
     "UserRole": "caseworker-civil-systemupdate",
     "CRUD": "CRUD"
   },
-  {
```
Comment (swalker125) [inline ccd-definition/AuthorisationCaseState/systemUpdate.json]: why is this removed?

Comment (pd-dudkiewicz) [reply ccd-definition/AuthorisationCaseState/systemUpdate.json]: This is to allow `Solicitor responds to defence` e2e test to pass locally while using idam stub. In idam stub we have one user with all roles, so when case moved to `STAYED` state, it was still available to user, instead of redirecting to Case List. We could change last test assertion if the stub is used, but I couldn't find a reason why we need this permission.

**Review by sabahirfan — DISMISSED** [review]

File: ccd-definition/AuthorisationCaseState/systemUpdate.json (hunk 1 above)
Comment (swalker125) [reply ccd-definition/AuthorisationCaseState/systemUpdate.json]: I don't think it's safe to remove, for example for my 202 story. CLAIMANT_RESPONSE event leaves case in stayed state from CCD definition, we then need to process events such as notifications etc. System update user needs to see case to trigger events

Comment (swalker125) [reply ccd-definition/AuthorisationCaseState/systemUpdate.json]: surely there are already notification events that need to happen already too?

**Review by sabahirfan — APPROVED** [review]
